src/utils.coffee

```diff
@@ -40,16 +48,13 @@ exports.mixpanelTrack = (event, properties = {}) ->
 
 	properties = _.cloneDeep(properties)
 
-	# Don't log private env vars (e.g. api keys)
-	if properties?.app?.env?
-		try
-			{ env } = properties.app
-			env = JSON.parse(env) if _.isString(env)
-			safeEnv = _.omit(env, config.privateAppEnvVars)
-			properties.app.env = JSON.stringify(safeEnv)
-		catch
-			properties.app.env = 'Fully hidden due to error in selective hiding'
-
+	# Filter properties to only send the whitelisted keys and values
+	properties = _.pick(properties, _.keys(mixpanelWhitelist))
+	properties = _.mapValues properties, (v, k) ->
```

Comment:
We should use something like https://github.com/nemtsov/json-mask This is a very ad-hoc implementation that will break once we have more recursion

Reply:
@petrosagg done (but I still kept the mask defined with an array, otherwise it was an unreadable long line)

(didn't know json-mask, it's nice!)